internal/hierarchy/hierarchy.go

```diff
@@ -48,6 +48,22 @@ func (p pathValue) Equal(o any) bool {
 	return true
 }
 
+// GetEndpoint returns the hierarchical path of endpoint.
+func GetEndpoint(endpoint resolver.Endpoint) []string {
```

Comment:
This name is incredibly misleading.  It's not getting an endpoint.  How about `FromEndpoint`?